util/platutil/platform.go

```diff
@@ -22,6 +22,11 @@ type Platform struct {
 	p      *specs.Platform
 }
 
+// IsPlatformDefined returns true when the platform was explicitly set
+func IsPlatformDefined(p Platform) bool {
+	return p.p != nil || p.user || p.native
```

Comment:
This doesn't seem right. I believe that in other places in the code we perform this check by doing `p == DefaultPlatform`.

Reply:
ah ha: https://github.com/earthly/earthly/blob/f50cd370e79d3f6b37b416c6020f1285201e0c06/builder/builder.go#L212

that looks better.